ENIGMAsystem/SHELL/Universal_System/image_formats.cpp

```diff
@@ -319,7 +319,8 @@ int image_save_png(string filename, const unsigned char* data, unsigned width, u
     file.write(reinterpret_cast<const char*>(buffer), std::streamsize(buffersize));
     file.close();
   }
-  delete[] buffer;
+  
+  free(buffer);
```

Comment:
If I'm being anal, it might be better to null these pointers out before passing them to libPNG, so they are never left dangling. I don't know if libPNG guarantees that it is safe to `free()` its pointer in an error case.

Reply:
https://github.com/enigma-dev/enigma-dev/blob/master/ENIGMAsystem/SHELL/Universal_System/lodepng.h#L155 It doesn't say to do anything other than free here.

Reply:
@JoshDreamland do not confuse lodepng and libpng, the latter is only a reference implementation.

@RemoveRusky What happens if lodepng has an error reading an image?

Reply:
the source is right there in our tree you can look yourself

Reply:
anyways do you just want unsigned char* buffer = nullptr; instead of unsigned char* buffer; or what?

Reply:
Yes, I think that's the safest thing. Just in case the LodePNG call returns without nulling the pointer itself.